hy/core/macros.hy

```diff
@@ -214,20 +214,42 @@
 
 (if-python2
   (defmacro/g! yield-from [expr]
-    `(do (import types)
-         (setv ~g!iter (iter ~expr))
-         (setv ~g!return nil)
-         (setv ~g!message nil)
-         (while true
-           (try (if (isinstance ~g!iter types.GeneratorType)
-                  (setv ~g!message (yield (.send ~g!iter ~g!message)))
-                  (setv ~g!message (yield (next ~g!iter))))
-           (except [~g!e StopIteration]
-             (do (setv ~g!return (if (hasattr ~g!e "value")
-                                     (. ~g!e value)
-                                     nil))
-               (break)))))
-           ~g!return))
+    `(do
+       (setv ~g!i (iter ~expr))
+       (try
+         (setv ~g!y (next ~g!i))
+         (except [~g!e StopIteration]
+                 (setv ~g!r (. ~g!e value)))
```

Comment:
This should be `(setv ~g!r (next (iter (. ~g!e args)) nil))`. StopIteration instances don't have the `value` attribute in Python 2.